indico/modules/events/editing/client/js/components/EditableTypeList.jsx

```diff
@@ -0,0 +1,148 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import editableTypesURL from 'indico-url:event_editing.api_editable_types';
+
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {Button, Checkbox, Form, Loader, Popup} from 'semantic-ui-react';
+import {Form as FinalForm} from 'react-final-form';
+import {Translate} from 'indico/react/i18n';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {FinalCheckbox, FinalSubmitButton, handleSubmitError} from 'indico/react/forms';
+import {indicoAxios} from 'indico/utils/axios';
+import {EditingType} from '../models';
+
+import './EditableTypeList.module.scss';
+
+export default function EditableTypeList({eventId}) {
+  const [editMode, setEditMode] = useState(false);
+  const {data: editableTypes, loading: isLoadingEditableTypes, reFetch} = useIndicoAxios({
+    url: editableTypesURL({confId: eventId}),
+    camelize: true,
+    trigger: eventId,
+  });
+
+  if (isLoadingEditableTypes) {
+    return <Loader inline="centered" active />;
+  } else if (!editableTypes) {
+    return null;
+  }
+
+  const handleSubmit = async formData => {
+    const url = editableTypesURL({confId: eventId});
+    const enabledTypes = Object.keys(formData).filter(name => formData[name]);
+    try {
+      await indicoAxios.post(url, {editable_types: enabledTypes});
+    } catch (e) {
+      return handleSubmitError(e);
+    }
+    reFetch();
+    setEditMode(false);
+  };
+
+  const renderEditMode = () => {
+    return (
+      <FinalForm
+        onSubmit={handleSubmit}
+        subscription={{submitting: true}}
+        initialValues={Object.assign(...editableTypes.map(([id, enabled]) => ({[id]: enabled})))}
```

Comment:
```suggestion
        initialValues={editableTypes.reduce((acc, [id, enabled]) => {
          acc[id] = enabled;
          return acc;
        }, {})}
```

Imo, this is more readable, but since spread has been widely used across other components and the latter is smaller, I'll leave this up to your criteria.